test/test-jsdomparser.js

```diff
@@ -0,0 +1,194 @@
+var path = require("path");
+var fs = require("fs");
+
+var chai = require("chai");
+chai.config.includeStack = true;
+var expect = chai.expect;
+
+// We want to load JSDOMParser, which isn't set up as commonjs libraries,
+// and so we need to do some hocus-pocus with 'vm' to import them on a separate scope
+// (identical) scope context.
+var vm = require("vm");
+var jsdomPath = path.join(__dirname, "..", "JSDOMParser.js");
+
+
+var scopeContext = {};
+// We generally expect dump() and console.{whatever} to work, so make these available
+// in the scope we're using:
+scopeContext.dump = console.log
+scopeContext.console = console;
+
+// Actually load file. NB: if the file has parse errors,
+// node is dumb and shows you a syntax error *at this callsite* . Don't try to find
+// a syntax error on this line, there isn't one. Go look in the file it's loading instead.
+vm.runInNewContext(fs.readFileSync(jsdomPath), scopeContext, jsdomPath);
+
+var JSDOMParser = scopeContext.JSDOMParser;
+
+var BASETESTCASE = '<html><body><p>Some text and <a class="someclass" href="#">a link</a></p>' +
+                   '<div id="foo">With a <script>With < fancy " characters in it because' +
+                   '</script> that is fun.<span>And another node to make it harder</span></div><form><input type="text"/><input type="number"/>Here\'s a form</form></body></html>';
+
+var baseDoc = new JSDOMParser().parse(BASETESTCASE);
+
+describe("Test JSDOM functionality", function() {
+  it("should work for basic operations with elements", function(done) {
+    expect(baseDoc.childNodes.length).eql(1);
+    expect(baseDoc.getElementsByTagName("*").length).eql(10);
+    var foo = baseDoc.getElementById("foo");
+    expect(foo.id).eql(foo.getAttribute("id"));
+    expect(foo.parentNode.localName).eql("body");
+    expect(baseDoc.body).eql(foo.parentNode);
+    expect(baseDoc.body.parentNode).eql(baseDoc.documentElement);
+    expect(baseDoc.body.childNodes.length).eql(3);
```

Comment:
How about increasing test granularity and organize the tests a little? That's usually helpful to get contextualized and descriptive information:

```js
describe("JSDOMParser", function() {
  describe("#parse()", function() {
    it("should return a Document", function() {
      expect(new JSDOMParser().parse(BASETESTCASE)).to.be.an.instanceOf(Document);
    });
  });
});

describe("Document", function() {
  var baseDoc;
  
  beforeEach(function() {
    baseDoc = new JSDOMParser().parse(BASETESTCASE);
  });

  describe("get childNodes()", function() {
    it("should return the expected number of child nodes", function() {
      expect(baseDoc.childNodes.length).eql(1);
    });
  });

  describe("#getElementsByTagName", function() {
    it("should return the expected number of tags", function() {
      expect(baseDoc.getElementsByTagName("*").length).eql(10);
    });
  });
  
  etc.